index.js

```diff
@@ -47,8 +47,10 @@ module.exports = (opts) => {
       }
 
       const sourceHttp2 = req.httpVersionMajor === 2
-      const headers = sourceHttp2 ? filterPseudoHeaders(req.headers) : req.headers
-      headers.host = url.hostname
+      const headers = { ...sourceHttp2 ? filterPseudoHeaders(req.headers) : req.headers }
+      headers['x-forwarded-host'] = req.headers.host
```

Comment:
FYI This broke http/2 support. There the host header is `:authority`.
It would be good to extract the `:authority` before or during the header normalization.

I got that error without adding the hostname explicitly onto the headers:
```js
{
    "statusCode": 500,
    "code": "ERR_HTTP_INVALID_HEADER_VALUE",
    "error": "Internal Server Error",
    "message": "Invalid value \"undefined\" for header \"x-forwarded-host\""
}
```